src/cpp/rtps/history/CacheChangePool.cpp

```diff
@@ -27,82 +27,99 @@
 
 
 namespace eprosima {
-namespace fastrtps{
+namespace fastrtps {
 namespace rtps {
 
 
 CacheChangePool::~CacheChangePool()
 {
-    logInfo(RTPS_UTILS,"ChangePool destructor");
+    logInfo(RTPS_UTILS, "ChangePool destructor");
     //Deletion process does not depend on the memory management policy
-    for(std::vector<CacheChange_t*>::iterator it = m_allCaches.begin();it!=m_allCaches.end();++it)
+    for (std::vector<CacheChange_t*>::iterator it = m_allCaches.begin(); it != m_allCaches.end(); ++it)
     {
         delete(*it);
     }
 }
 
-CacheChangePool::CacheChangePool(int32_t pool_size, uint32_t payload_size, int32_t max_pool_size, MemoryManagementPolicy_t memoryPolicy) :
-    memoryMode(memoryPolicy)
+CacheChangePool::CacheChangePool(
+        int32_t pool_size,
+        uint32_t payload_size,
+        int32_t max_pool_size,
+        MemoryManagementPolicy_t memoryPolicy)
+    : memoryMode(memoryPolicy)
 {
     //Common for all modes: Set the payload size (maximum allowed), size and size limit
     ++pool_size;
-    logInfo(RTPS_UTILS,"Creating CacheChangePool of size: "<< pool_size << " with payload of size: " << payload_size);
+    logInfo(RTPS_UTILS,
+            "Creating CacheChangePool of size: " << pool_size << " with payload of size: " << payload_size);
 
     m_payload_size = payload_size;
     m_initial_payload_size = payload_size;
     m_pool_size = 0;
-    if(max_pool_size > 0)
+    if (max_pool_size > 0)
     {
         if (pool_size > max_pool_size)
         {
             m_max_pool_size = (uint32_t)abs(pool_size);
         }
         else
-            m_max_pool_size = (uint32_t)abs(max_pool_size + 1);
+        {
+            m_max_pool_size = (uint32_t)abs(max_pool_size /* + 1*/);
```

Comment:
After checking that it correctly works, we are safe to remove the `+1`, instead of leaving it commented out.

```suggestion
            m_max_pool_size = (uint32_t)abs(max_pool_size);
```

Reply:
I leave it commented out because I don't know if it may have any additional side effects with other QoS/Topic configurations. It is strange that didn't arise before.